src/Experimental/ExperimentRegistry.php

```diff
@@ -0,0 +1,220 @@
+	/**
+	 * The registered experiments.
+	 *
+	 * @var ?array<string,class-string<\WPGraphQL\Experimental\Experiment\AbstractExperiment>>
+	 */
+	protected static $registry;
+
+	/**
+	 * The initialized experiments.
+	 *
+	 * @var ?array<string,\WPGraphQL\Experimental\Experiment\AbstractExperiment>
+	 */
+	protected static $experiments;
+
+	/**
+	 * The active experiments.
+	 *
+	 * @var ?array<string,\WPGraphQL\Experimental\Experiment\AbstractExperiment>
+	 */
+	protected static $active_experiments;
+
+	/**
+	 * Initializes the Experimental Functionality for WPGraphQL
+	 */
+	public function init(): void {
+		$this->register_experiments();
+		$this->load_experiments();
+	}
+
+	/**
+	 * Whether the current user can manage experimental features.
+	 *
+	 * @uses 'graphql_experimental_features_cap' filter to determine the capability required.
+	 */
+	public static function can_manage_experiments(): bool {
+		return current_user_can( self::capability() );
+	}
+
+	/**
+	 * The capability required to turn experimental features on and off.
+	 *
+	 * Defaults to `manage_options`.
+	 */
+	public static function capability(): string {
+		/**
+		 * Filters the capability required to turn experimental features on and off.
+		 *
+		 * @param string $capability The capability required to turn experimental features on and off. Defaults to `manage_options`.
+		 */
+		return apply_filters( 'graphql_experimental_features_cap', 'manage_options' );
+	}
+
+	/**
+	 * Checks whether the experiment is enabled.
+	 *
+	 * @return array<string,class-string<\WPGraphQL\Experimental\Experiment\AbstractExperiment>>
+	 */
+	public static function get_experiment_registry(): array {
+		if ( ! isset( self::$registry ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'Registered experiments have not been set. Make sure not to call this function before the `graphql_experiments_registered` hook.', 'wp-graphql' ),
+				'@todo'
+			);
+
+			return [];
+		}
+
+		return self::$registry;
+	}
+
+	/**
+	 * Gets the list of all experiments.
+	 *
+	 * @return array<string,\WPGraphQL\Experimental\Experiment\AbstractExperiment>
+	 */
+	public static function get_experiments(): array {
+		if ( ! isset( self::$experiments ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'Experiments have not been loaded. Make sure not to call this function before the `graphql_experiments_loaded` hook.', 'wp-graphql' ),
+				'@todo'
+			);
+
+			return [];
+		}
+
+		return self::$experiments;
+	}
+
+	/**
+	 * Get the list of active experiments.
+	 *
+	 * @return array<string,\WPGraphQL\Experimental\Experiment\AbstractExperiment>
+	 */
+	public static function get_active_experiments(): array {
+		if ( ! isset( self::$active_experiments ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'Active experiments have not been loaded. Make sure not to call this function before the `graphql_experiments_loaded` hook.', 'wp-graphql' ),
+				'@todo'
+			);
+
+			return [];
+		}
+
```

Comment:
if multiple experiments are registered with the same slug we should call that out. 

When testing the API, I accidentally left the slug the same as the test experiement, and my new one showed up but the test one disappeared and it took me a second to realize that it was because I left the slug the same. I caught it quickly, but would have caught it quicker if something warned me about the duplicate slugs.